cmd/kubeadm/app/cmd/upgrade/diff_test.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+const (
+	testUpgradeDiffConfig = `apiVersion: kubeadm.k8s.io/v1alpha1
+kind: MasterConfiguration
+kubernetesVersion: 1.11.0
+`
+	testUpgradeDiffManifest = `some-empty-file-to-diff`
+)
+
+func TestRunDiff(t *testing.T) {
+	// create temp directory
+	tmpDir, err := ioutil.TempDir("", "kubeadm-upgrade-diff-test")
+	if err != nil {
+		t.Errorf("Unable to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// write a config file
+	validConfigFilePath := filepath.Join(tmpDir, "test-config-file")
+	cfgFile, err := os.Create(validConfigFilePath)
+	if err != nil {
+		t.Errorf("Unable to create file %q: %v", validConfigFilePath, err)
+	}
+	defer cfgFile.Close()
+	if _, err = cfgFile.WriteString(testUpgradeDiffConfig); err != nil {
```

Comment:
it might be easier to just use a `testdata` directory so you don't need to do all this setup and teardown